yaml/secret.go

```diff
@@ -89,6 +90,29 @@ func (s *SecretSlice) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		if secret.Origin.Empty() && len(secret.Type) == 0 {
 			secret.Type = constants.SecretRepo
 		}
+
+		// implicitly set `pull` field if empty
+		if len(secret.Origin.Pull) == 0 {
+			secret.Origin.Pull = constants.PullNotPresent
+		}
```

Comment:
Set a default pull policy of `not_present` for the `github.com/go-vela/types/yaml.Origin` type